packages/vite/src/node/utils.ts

```diff
@@ -552,11 +552,12 @@ export async function processSrcSet(
   const imageCandidates: ImageCandidate[] = srcs
     .split(',')
     .map((s) => {
-      const [url, descriptor] = s
+      const [url, ...descriptorArr] = s
         .replace(escapedSpaceCharacters, ' ')
         .trim()
-        .split(' ', 2)
-      return { url, descriptor }
+        .split(' ')
```

Comment:
Should we split after `)`, `'`, `"` here? What happens if we have something like `var(  --my-var)`?

Reply:
Thanks for review!
Yes, the space inside `var()` or `url()` will cause some trouble.Maybe I can clean space inside it.

Reply:
Thank you for pushing forward with a fix! ;)
Maybe a regex is better to match the first part? var(), url(), '...', "..." ?

Reply:
Update.
According to [spec](https://drafts.csswg.org/css-images-4/#funcdef-image-set-type), support 2 condition.
1. function call like `var()`, `url()`, `linear-gradient()`
2. url like `'test.png'`, `"test.png"`